src/pip/_internal/vcs/versioncontrol.py

```diff
@@ -82,6 +82,7 @@ def __init__(
         self.extra_args = extra_args
         self.rev = rev
         self.vc_class = vc_class
+        self.branch_name = None  # type: Optional[str]
```

Comment:
I would keep it above `self.extra_args` so they are sorted, but also second-guess myself because it's reasonable to assign from parameters to properties first and set other members afterwards.

Reply:
I usually group instance attributes set from constants separately because it’s easier visually to match the attributes set from arguments with the corresponding arguments. It also helps conceptually because the constant ones are going by different “rules” and so require different thinking (and so they stand out better). But happy to reorder if that’s what you like more.

Reply:
I'm OK with it as-is. Keeping them ordered might be more approachable or easier to enforce, but would probably fail to scale after just a few more, like [here](https://github.com/pypa/pip/blob/76ae377d7e38b5df8587d63b52464a08471eb1d1/src/pip/_internal/req/req_install.py#L90-L165).

Reply:
Okay, thanks. (One thing I always go back and forth on by the way is whether to order the attributes alphabetically or by parameter order. The former makes it harder to check visually, but the former also has its advantages...)